### PR TITLE
Bovdfixes

### PR DESCRIPTION
**Pull request by myrlyn**

### CODE REVIEW

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_abilitycategories.lst [hunk 1]
```diff
@@ -22,7 +22,7 @@ ABILITYCATEGORY:Thrall of Orcus Class Feature			VISIBLE:QUALIFY	EDITABLE:YES	EDI
 ABILITYCATEGORY:Ur-Priest Class Feature				VISIBLE:QUALIFY	EDITABLE:YES	EDITPOOL:NO						CATEGORY:Special Ability	TYPE:Ur-Priest Class Feature				PLURAL:Ur-Priest Class Features				DISPLAYLOCATION:Class Features
 ABILITYCATEGORY:Vermin Lord Class Feature				VISIBLE:QUALIFY	EDITABLE:YES	EDITPOOL:NO						CATEGORY:Special Ability	TYPE:Vermin Lord Class Feature			PLURAL:Vermin Lord Class Features				DISPLAYLOCATION:Class Features
 ABILITYCATEGORY:Warrior of Darkness Class Feature		VISIBLE:QUALIFY	EDITABLE:YES	EDITPOOL:NO						CATEGORY:Special Ability	TYPE:Warrior of Darkness Class Feature		PLURAL:Warrior of Darkness Class Features			DISPLAYLOCATION:Class Features
-
+ABILITYCATEGORY:Thrall of Orcus Choice					VISIBLE:QUALIFY	EDITABLE:YES	EDITPOOL:YES					CATEGORY:Special Ability	TYPE:OrcusChoice									PLURAL:Thrall of Orcus Choices		DISPLAYLOCATION:Class Features
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_abilitycategories.lst]: New ability category to handle per-level choice of Bonus Feat or Spells for Thrall of orcus

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst [hunk 2]
```diff
@@ -69,16 +69,8 @@ CLASS:Diabolist	PREALIGN:LE	PRECHECKBASE:1,Will=5	PREABILITY:2,CATEGORY=FEAT,Evi
...
-2	ADD:SPELLCASTER|ANY
-3	ADD:SPELLCASTER|ANY
-4	ADD:SPELLCASTER|ANY
-5	ADD:SPELLCASTER|ANY
-6	ADD:SPELLCASTER|ANY
-7	ADD:SPELLCASTER|ANY
-8	ADD:SPELLCASTER|ANY
-9	ADD:SPELLCASTER|ANY
-10	ADD:SPELLCASTER|ANY
+1:REPEATLEVEL:1	ADD:SPELLCASTER|ANY
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Prefer formulae for spellcaster progression

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst [hunk 3]
```diff
@@ -95,11 +87,7 @@ CLASS:Disciple of Asmodeus	PREALIGN:LE,NE	PREABILITY:3,CATEGORY=FEAT,Disciple of
 # Class Name			Skill Pts/Lvl	Class Skill
 CLASS:Disciple of Asmodeus	STARTSKILLPTS:4	CSKILL:Bluff|Concentration|Decipher Script|Diplomacy|Gather Information|Hide|Intimidate|TYPE=Knowledge|Listen|Search|Sense Motive|Spellcraft|Spot
 ###Block
-1	ADD:SPELLCASTER|ANY
-3	ADD:SPELLCASTER|ANY
-5	ADD:SPELLCASTER|ANY
-7	ADD:SPELLCASTER|ANY
-9	ADD:SPELLCASTER|ANY
+1:REPEATLEVEL:2	ADD:SPELLCASTER|ANY
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Prefer formulae for spellcaster progression

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst [hunk 4]
```diff
@@ -310,11 +298,7 @@ CLASS:Thrall of Graz'zt	PREALIGN:LE,NE,CE	PREABILITY:2,CATEGORY=FEAT,Thrall to D
 # Class Name		Skill Pts/Lvl	Class Skill
 CLASS:Thrall of Graz'zt	STARTSKILLPTS:4	CSKILL:Bluff|Concentration|Decipher Script|Diplomacy|Gather Information|Hide|TYPE=Knowledge|Listen|Move Silently|Open Lock|TYPE=Perform|Search|Sense Motive|Sleight of Hand|Spellcraft|Spot|Survival|Use Magic Device
 ###Block
-1	ADD:SPELLCASTER|ANY
-3	ADD:SPELLCASTER|ANY
-5	ADD:SPELLCASTER|ANY
-7	ADD:SPELLCASTER|ANY
-9	ADD:SPELLCASTER|ANY
+1:REPEATLEVEL:2	ADD:SPELLCASTER|ANY
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Prefer formulae for spellcaster progression

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst [hunk 5]
```diff
@@ -407,11 +392,7 @@ CLASS:Vermin Lord	PREALIGN:LE,NE,CE	PREABILITY:1,CATEGORY=FEAT,Verminfriend	PRES
 # Class Name	Skill Pts/Lvl	Class Skill
 CLASS:Vermin Lord	STARTSKILLPTS:4	CSKILL:Bluff|Climb|Concentration|Hide|Knowledge (Nature)|Listen|Move Silently|TYPE=Perform|Sleight of Hand|Search|Spellcraft|Spot
 ###Block
-1	ADD:SPELLCASTER|ANY
-3	ADD:SPELLCASTER|ANY
-5	ADD:SPELLCASTER|ANY
-7	ADD:SPELLCASTER|ANY
-9	ADD:SPELLCASTER|ANY
+1:REPEATLEVEL:2	ADD:SPELLCASTER|ANY
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Prefer formulae for spellcaster progression

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst [hunk 6]
```diff
@@ -359,6 +343,7 @@ CLASS:Thrall of Orcus	STARTSKILLPTS:2	CSKILL:Bluff|Concentration|TYPE=Craft|Hide
 ###Block
 1	ABILITY:Thrall of Orcus Class Feature|AUTOMATIC|Thrall of Orcus ~ Weapon and Armor Proficiency
 1	ABILITY:Thrall of Orcus Class Feature|AUTOMATIC|Thrall of Orcus ~ Bonus Feat or Spells
+1:REPEATLEVEL:3	BONUS:ABILITYPOOL|Thrall of Orcus Choice|1
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Provide choices for Thrall of Orcus for Spells or Bonus Feats

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst [hunk 7]
```diff
@@ -3,15 +3,16 @@ SOURCELONG:Book of Vile Darkness	SOURCESHORT:BV	SOURCEDATE:2002-10
...
-FOLLOWER:CancerousCompanionLVL=5	TYPE:Cancer	ABILITY:Special Ability|AUTOMATIC|Uncanny Dodge ~ Base														DEFINESTAT:MINVALUE|INT|9	BONUS:VAR|UncannyDodgeFlankingLevel|CancerousCompanionLVL	BONUS:VAR|UncannyDodgeLevel|1
-FOLLOWER:CancerousCompanionLVL=6	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|10
-FOLLOWER:CancerousCompanionLVL=7	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|11
-FOLLOWER:CancerousCompanionLVL=8	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|12	BONUS:VAR|UncannyDodgeLevel|1
-FOLLOWER:CancerousCompanionLVL=9	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|13
+FOLLOWER:CancerousCompanionLVL=2	TYPE:Cancer	ABILITY:Special Ability|AUTOMATIC|Animal Companion ~ Link		COPYMASTERBAB:MASTER	COPYMASTERCHECK:MASTER	USEMASTERSKILL:YES	DEFINESTAT:MINVALUE|INT|6				DEFINESTAT:MINVALUE|STR|mastervar("STRSCORE")	DEFINESTAT:MINVALUE|DEX|mastervar("DEXSCORE")	DEFINESTAT:MINVALUE|CON|mastervar("CONSCORE")		DEFINESTAT:MINVALUE|WIS|mastervar("WISSCORE")		DEFINESTAT:MINVALUE|CHA|mastervar("CHASCORE")											VISION:Blindsight (30')
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst]: Updated Cancerous Companion to correctly apply abilities

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst [hunk 8]
```diff
@@ -3,15 +3,16 @@ SOURCELONG:Book of Vile Darkness	SOURCESHORT:BV	SOURCEDATE:2002-10
...
+FOLLOWER:CancerousCompanionLVL=7	TYPE:Cancer	BONUS:VAR|MasterSpellTimes|2																							DEFINESTAT:MINVALUE|INT|11
+FOLLOWER:CancerousCompanionLVL=8	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|12
+FOLLOWER:CancerousCompanionLVL=9	TYPE:Cancer	SAB:Can form a 1-ft. tendril of flesh capable of making touch attacks, including the cancer mage’s contagion and poison spelllike abilities.																							DEFINESTAT:MINVALUE|INT|13
 FOLLOWER:CancerousCompanionLVL=10	TYPE:Cancer																								DEFINESTAT:MINVALUE|INT|14																VISION:Blindsight (60')
+MASTERBONUSRACE:Cancer				BONUS:VAR|UncannyDodgeLevel|1|PREVARGTEQ:CancerousCompanionLVL,8	BONUS:VAR|UncannyDodgeFlankingLevel|CancerousCompanionLVL|PREVARGTEQ:CancerousCompanionLVL,5	BONUS:VAR|UncannyDodgeLevel|1|PREVARGTEQ:CancerousCompanionLVL,5	TYPE:Cancer	ABILITY:Special Ability|AUTOMATIC|Rogue ~ Evasion|PREVARGTEQ:CancerousCompanionLVL,4		ABILITY:Special Ability|AUTOMATIC|Uncanny Dodge ~ Base|PREVARGTEQ:CancerousCompanionLVL,5
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst]: added a race entry for use in Cancerous Companion

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_feats.lst [hunk 9]
```diff
@@ -4,32 +4,33 @@ SOURCELONG:Book of Vile Darkness	SOURCESHORT:BV	SOURCEDATE:2002-10
...
-Disciple of Darkness										CATEGORY:FEAT	TYPE:Vile.DemonicAllegiance	VISIBLE:YES	PREALIGN:LE,NE,CE											!PREABILITY:1,CATEGORY=FEAT,TYPE=DemonicAllegiance																							DEFINE:AuraEvil|0								DESC:See Text																					ABILITY:Special Ability|VIRTUAL|Aura of Evil																																ADDSPELLLEVEL:1																															BONUS:VAR|AuraEvil|TL																											SOURCEPAGE:p.
+Boost Spell Resistance										CATEGORY:FEAT	TYPE:General			VISIBLE:YES	PREALIGN:LE,NE,CE																																																	DESC:See Text																																																																				BONUS:MISC|SR|2|TYPE=Profane																																																					SOURCEPAGE:p.
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_feats.lst]: Several missing rules tags fixed

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_races.lst [hunk 10]
```diff
@@ -26,3 +26,4 @@ Kython (Slaymaster)	OUTPUTNAME:Kython [NAME]										STARTFEATS:1	SIZE:L	MOVE:W
...
 Vilewight																	STARTFEATS:1	SIZE:M	MOVE:Walk,30		FACE:5	REACH:5	VISION:Darkvision (60')	PREALIGN:NE																				BONUS:COMBAT|AC|6|TYPE=NaturalArmor	BONUS:SKILL|Move Silently|8|TYPE=Racial						BONUS:STAT|STR|8	BONUS:STAT|DEX|4	BONUS:STAT|INT|17	BONUS:STAT|WIS|12	BONUS:STAT|CHA|4				BONUS:VAR|EnergyDrainLevels|2	MONCSKILL:Climb|Hide|Knowledge (arcana)|Listen|Move Silently|Spot																																											ABILITY:Internal|AUTOMATIC|Race Traits ~ Vilewight																																																																							LEGS:2	HANDS:2	NATURALATTACKS:Claw,Weapon.Natural.Melee.Finesseable.Piercing.Slashing,*2,1d6|Bite,Weapon.Natural.Finesseable.Melee.Piercing.Slashing,*1,1d8|Tendril,Weapon.Natural.Finesseable.Melee.Piercing.Slashing,*1,1d6													MONSTERCLASS:Undead:12				RACETYPE:Undead						TYPE:Undead																CR:8			SOURCEPAGE:BoVD
+Cancer																		STARTFEATS:0	SIZE:M	MOVE:Walk,0			FACE:0	REACH:0	BONUS:STAT|INT|-4
```
Comment (myrlyn) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_races.lst]: New race to be selected as a Cancerous Companion

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst (hunk 6 above)
Comment (BahamutDragon) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_classes.lst]: Out of standard convention. BONUS should be in the class feature with the formula based off the class level.

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst (hunk 8 above)
Comment (BahamutDragon) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_companionmods.lst]: Why are we using SAB? Those should be additional abilities.

File: data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_feats.lst [hunk 11]
```diff
@@ -4,32 +4,33 @@ SOURCELONG:Book of Vile Darkness	SOURCESHORT:BV	SOURCEDATE:2002-10
...
+Dark Speech												CATEGORY:FEAT	TYPE:Vile				VISIBLE:YES	PREALIGN:LE,NE,CE	PRECHECKBASE:1,Will=5													PRESTAT:2,INT=15,CHA=15																			DEFINE:AuraEvil|0								DESC:See Chapter 2 of Book of Vile Darkness.																					ABILITY:Special Ability|VIRTUAL|Aura of Evil																																																																	BONUS:VAR|AuraEvil|TL																											SOURCEPAGE:p.
+Deformity (Clawed Hands)									CATEGORY:FEAT	TYPE:Vile				VISIBLE:YES	PREALIGN:LE,NE,CE		ABILITY:Special Ability|AUTOMATIC|Clawed Hand			PREABILITY:1,CATEGORY=FEAT,Willing Deformity																														DEFINE:AuraEvil|0								DESC:Gaint a claw attack.														ABILITY:Special Ability|VIRTUAL|Aura of Evil																																																																	BONUS:VAR|AuraEvil|TL																											SOURCEPAGE:p.
```
Comment (BahamutDragon) [inline data/35e/wizards_of_the_coast/supplement/book_of_vile_darkness/bv_feats.lst]: Typo "Gaint a claw attack". Think that was intended to be 'Grant'

**Comment by BahamutDragon** [comment]
Not usual to have line comments initiated by the PR opener. That could have been summarized in the initial Comment section.

**Comment by BahamutDragon** [comment]
Not sure if you intend to assist with the conversion effort to the next development version. That is why I am attempting to stick with the standards to make conversion easier on myself.

**Pull request merged**